components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/LabelDAOImpl.java

```diff
@@ -191,129 +92,174 @@ private static void insertAccessUrlMappings(String labelId, List<String> accessU
 
-                for (int i = 0; i < labelIDs.size(); ++i) {
-                    statement.setString(i + 1, labelIDs.get(i));
-                }
-
-                try (ResultSet rs = statement.executeQuery()) {
-                    while (rs.next()) {
-                        labelNames.add(rs.getString("NAME"));
-                    }
+                for (String accessUrl : accessUrls) {
+                    statement.setString(1, labelId);
+                    statement.setString(2, accessUrl);
+                    statement.addBatch();
                 }
+                statement.executeBatch();
+            } catch (SQLException e) {
+                String message = "Error while adding access url mappings for [label id] " + labelId;
+                log.error(message, e);
+                throw new APIMgtDAOException(e);
             }
         }
-        return labelNames;
     }
 
     /**
-     * Retrieve label id by label name
-     *
-     * @param labelName Name of the label
-     * @return Id of the label
-     * @throws SQLException if error occurs while retrieving label id
+     * @see LabelDAO#updateLabel(Label)
      */
-    static String getLabelID(String labelName) throws SQLException {
-
-        final String query = "SELECT LABEL_ID from AM_LABELS where NAME=?";
+    @Override
+    public Label updateLabel(Label updatedLabel) throws APIMgtDAOException {
+        final String query = "UPDATE AM_LABELS SET NAME=? , TYPE_NAME=? WHERE LABEL_ID=?";
+        String labelId = updatedLabel.getId();
+        List<String> accessURLs = updatedLabel.getAccessUrls();
 
         try (Connection connection = DAOUtil.getConnection();
              PreparedStatement statement = connection.prepareStatement(query)) {
-            statement.setString(1, labelName);
-            statement.execute();
-
-            try (ResultSet rs = statement.getResultSet()) {
-                if (rs.next()) {
-                    return rs.getString("LABEL_ID");
+            try {
+                 connection.setAutoCommit(false);
+                 statement.setString(1, updatedLabel.getName());
+                 if (updatedLabel.getType() != null) {
+                     statement.setString(2, updatedLabel.getType().toUpperCase(Locale.ENGLISH));
+                 } else {
+                     statement.setString(2, updatedLabel.getType());
+                 }
+                 statement.setString(3, updatedLabel.getId());
+                 statement.executeUpdate();
+                 connection.commit();
+                 deleteLabelAccessUrlMappings(labelId);
+                if ("GATEWAY".equalsIgnoreCase(updatedLabel.getType())) {
+                   insertAccessUrlMappings(labelId, accessURLs);
+                } else {
+                    accessURLs = new ArrayList<>();
                 }
+            } catch (SQLException e) {
+                connection.rollback();
+                String message = "Error while updating the label" +
```

Comment:
now we no need it as we chose to log in topmost layer

Reply:
done